libraries/rush-lib/src/logic/operations/OperationMetadataManager.ts

```diff
@@ -112,31 +142,48 @@ export class OperationMetadataManager {
 
   public async tryRestoreAsync({
     terminal,
-    logPath,
+    terminalProvider,
     errorLogPath
   }: {
+    terminalProvider: ITerminalProvider;
     terminal: ITerminal;
-    logPath: string;
     errorLogPath: string;
   }): Promise<void> {
     await this.stateFile.tryRestoreAsync();
 
-    // Append cached log into current log file
-    terminal.writeLine(`Restoring cached log file at ${this._logPath}`);
     let logReadStream: fs.ReadStream | undefined;
     try {
-      logReadStream = fs.createReadStream(this._logPath, {
-        encoding: 'utf-8'
-      });
-      for await (const data of logReadStream) {
-        terminal.write(data);
+      if (await FileSystem.existsAsync(this._logChunksPath)) {
+        logReadStream = fs.createReadStream(this._logChunksPath, {
+          encoding: 'utf-8'
+        });
+        const processing: Chain = chain([
+          logReadStream,
+          parser(),
+          function process({ value: chunk }: { value: ITerminalChunk }) {
```

Comment:
Any reason not to use a lambda expression here?

Reply:
Nope, updated in [e8b979c](https://github.com/microsoft/rushstack/pull/4668/commits/e8b979ccdd5f37686ce3b576c35bdc0511c7544d)